doc/api/events.markdown

```diff
@@ -377,6 +377,42 @@ listener array. If any single listener has been added multiple times to the
 listener array for the specified `event`, then `removeListener` must be called
 multiple times to remove each instance.
 
+Note that once an event has been emitted, all listeners attached to it at the
+time of emitting will be called in order. This implies that any `removeListener`
+call *after* emitting and *before* the last listener finishes execution will
+not affect the current listener array. Subsequent events will behave as expected.
+
+```js
+const myEmitter = new MyEmitter();
```

Comment:
`const EventEmitter = require('events');
class MyEmitter extends EventEmitter {}`

This initialization has been done in the first example of events doc and from there on each example simply uses _myEmitter_ . I was trying to be consistent with the existing doc. 
As for the linting of code blocks in docs, I'll do that at once.